docker/models/containers.py

```diff
@@ -709,7 +716,14 @@ def run(self, image, command=None, stdout=True, stderr=False,
         if exit_status != 0:
             stdout = False
             stderr = True
-        out = container.logs(stdout=stdout, stderr=stderr)
+
+        logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
+
+        if logging_driver == 'json-file' or logging_driver == 'journald':
```

Comment:
I don't think this is the right approach; docker now supports logging driver plugins, which may (or may not) support reading. Hard-coding these options here will block that option.

the `500` error returned does not look right to me though, I think that should be fixed